modules/react-mapbox/src/components/source.ts

```diff
@@ -83,17 +83,16 @@ export function Source(props: SourceProps) {
   const [, setStyleLoaded] = useState(0);
 
   const id = useMemo(() => props.id || `jsx-source-${sourceCounter++}`, []);
+  const forceUpdate = useCallback(() => setStyleLoaded(version => version + 1), []);
 
   useEffect(() => {
     if (map) {
-      /* global setTimeout */
-      const forceUpdate = () => setTimeout(() => setStyleLoaded(version => version + 1), 0);
-      map.on('load', forceUpdate);
+      // Fired on initial load signaling the map is ready to add custom sources
+      // Subsequently fired on style changes
       map.on('styledata', forceUpdate);
       forceUpdate();
 
```

Comment:
Safe to remove since styledata fires at the same time?

Reply:
`styledata` is the "correct" event to observe for map style changes. `sourcedata` fires much more frequently (e.g. on every new tile load) and is excessive once source is added.

In this PR, we subscribe to both `styledata` and `sourcedata` while the map is loading - either event could lead to `map.isStyleLoaded()` become true - once map is loaded we add the source, and unsubscribe from `sourcedata`. If the user swaps out the map stylesheet, `styledata` will fire and `sourcedata` is added again.